helm/nessie/templates/_helpers.tpl

```diff
@@ -291,50 +291,65 @@ Define environkent variables for catalog storage options.
-{{- include "nessie.secretToEnv" (list .Values.catalog.storage.adls.defaultOptions.sasTokenSecret "sasToken" "nessie.catalog.service.adls.default-options.sas-token" . ) }}
+{{ include "nessie.secretToEnv" (list .Values.catalog.storage.adls.defaultOptions.accountSecret "accountName" "adls.default-options.account" "name" true . ) }}
+{{- include "nessie.secretToEnv" (list .Values.catalog.storage.adls.defaultOptions.accountSecret "accountKey" "adls.default-options.account" "secret" false . ) }}
+{{- include "nessie.secretToEnv" (list .Values.catalog.storage.adls.defaultOptions.sasTokenSecret "sasToken" "adls.default-options.sas-token" "token" true . ) }}
 {{- range $i, $filesystem := .Values.catalog.storage.adls.filesystems -}}
 {{- with $global }}
-{{- include "nessie.secretToEnv" (list $filesystem.accountSecret "accountName" (printf "nessie.catalog.service.adls.file-systems.filesystem%d.account.name" (add $i 1)) . ) }}
-{{- include "nessie.secretToEnv" (list $filesystem.accountSecret "accountKey" (printf "nessie.catalog.service.adls.file-systems.filesystem%d.account.secret" (add $i 1)) . ) }}
-{{- include "nessie.secretToEnv" (list $filesystem.sasTokenSecret "sasToken" (printf "nessie.catalog.service.adls.file-systems.filesystem%d.sas-token" (add $i 1)) . ) }}
+{{- include "nessie.secretToEnv" (list $filesystem.accountSecret "accountName" (printf "adls.file-systems.filesystem%d.account" (add $i 1)) "name" true . ) }}
+{{- include "nessie.secretToEnv" (list $filesystem.accountSecret "accountKey" (printf "adls.file-systems.filesystem%d.account" (add $i 1)) "secret" false . ) }}
+{{- include "nessie.secretToEnv" (list $filesystem.sasTokenSecret "sasToken" (printf "adls.file-systems.filesystem%d.sas-token" (add $i 1)) "token" true . ) }}
 {{- end -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
 Define an env var from secret key.
+
+Secrets are (can be) composite values - think of a username+password.
+Secrets are not (no longer) present (or directly resolvable) from the bucket option types, but have to be resolved
+via a symbolic name, which is something like 'nessie-catalog-secrets.s3.default-options.access-key'. The bucket
+config types know about that symbolic name and resolve it via a SecretsProvider, which resolves via Quarkus' config.
+
 */}}
 {{- define "nessie.secretToEnv" -}}
 {{- $secret := index . 0 -}}
 {{- $key := index . 1 -}}
-{{- $envVarName := index . 2 -}}
-{{- $global := index . 3 -}}
+{{- $midfix := index . 2 -}}
+{{- $suffix := index . 3 -}}
+{{- $addRef := index . 4 -}}
+{{- $global := index . 5 -}}
 {{- if $secret -}}
 {{- $secretName := get $secret "name" -}}
 {{- $secretKey := get $secret $key -}}
 {{- with $global -}}
 {{- if (and $secretName $secretKey) -}}
-- name: {{ $envVarName | quote }}
+{{ if $addRef -}}
+#
+# {{ $midfix }}
+#
+- name: {{ (printf "nessie.catalog.service.%s" $midfix) | quote }}
+  value: {{ (printf "urn:nessie-secret:quarkus:nessie-catalog-secrets.%s" $midfix) | quote }}
```

Comment:
All non-secret config goes to the `ConfigMap`, but this still produces and env. var for the `nessie` pod:
```
          env:
            #
            # s3.default-options.access-key
            #
            - name: "nessie.catalog.service.s3.default-options.access-key"
              value: "urn:nessie-secret:quarkus:nessie-catalog-secrets.s3.default-options.access-key"
 ```
Why not keep it in the `ConfigMap`?